tests/baselines/reference/strictOptionalProperties1.errors.txt

```diff
@@ -135,15 +138,24 @@ strictOptionalProperties1.ts(211,1): error TS2322: Type 'string | boolean | unde
         t = [42, 'abc'];
         t = [42, 'abc', true];
         t = [42, ,];
+        ~
+!!! error TS2322: Type '[number, undefined]' is not assignable to type '[number, string?, boolean?]'.
+!!! error TS2322:   Type at position 1 in source is not compatible with type at position 1 in target.
+!!! error TS2322:     Type 'undefined' is not assignable to type 'string'.
         t = [42, , ,];
+        ~
+!!! error TS2322: Type '[number, undefined, undefined]' is not assignable to type '[number, string?, boolean?]'.
+!!! error TS2322:   Type at position 1 in source is not compatible with type at position 1 in target.
+!!! error TS2322:     Type 'undefined' is not assignable to type 'string'.
         t = [42, , , ,];  // Error
         ~
-!!! error TS2322: Type '[number, never?, never?, never?]' is not assignable to type '[number, string?, boolean?]'.
-!!! error TS2322:   Target allows only 3 element(s) but source may have more.
+!!! error TS2322: Type '[number, undefined, undefined, undefined]' is not assignable to type '[number, string?, boolean?]'.

```

Comment:
Currently this is how it works:
```ts
const arr = [10, ,] as const; // const arr: readonly [10, never?]
```

I think that this is an improvement though. For the end user, it feels much more readable with `undefined` here for the omitted expression than `never` - especially within the error message. I could likely bring back `never` here though - it's assignable to everything after all.

Reply:
Note that `never` only appears here with EOPT and not without it. Should those really behave differently in this regard? I'd love to learn about some compelling example of it being useful.

Reply:
The optionality that was added to all elements since the omitted expression in the array literal breaks cases like this:
```ts
// @strict: true
// @exactOptionalPropertyTypes: true

function test3(arg: [number, undefined, string]) {}
test3([10, , "foo"]); // error but it should be OK
```